src/Core/Configurations/RuntimeConfigValidator.cs

```diff
@@ -273,8 +271,11 @@ public static void ValidateGlobalEndpointRouteConfig(RuntimeConfig runtimeConfig
                     subStatusCode: DataApiBuilderException.SubStatusCodes.ConfigValidationError);
             }
 
-            ValidateRestPathForRelationalDbs(runtimeConfig);
-            ValidateGraphQLPath(runtimeConfig);
+            string? runtimeBaseRoute = runtimeConfig.Runtime.BaseRoute;
+            ValidateURIComponent(runtimeBaseRoute, RuntimeOptions.PROPERTY_NAME_BASE_ROUTE, ApiType.All);
+
+            ValidateRestURI(runtimeConfig);
+            ValidateGraphQLURI(runtimeConfig);
```

Comment:
To avoid having to created extra properties on `RuntimeOptions`, and passing around property names and Api types, maybe this spot could be refactored to a more generic `IsValidUri(uri)`, if false, raise an exception and statically create the message with "REST path invalid" etc. 

These are essentially the same. Could you add any cosmos checks prior to validating the rest uri?
- ValidateRestURI(runtimeConfig);
- ValidateGraphQLURI(runtimeConfig);

Any exceptions thrown in validation helpers like ValidateRestUri can instead be logged with ILogger, the whole point being that the error is surfaced to the user on the console.

Reply:
Incorporated.